app/lib/actions/katello/host/attach_subscriptions.rb

```diff
@@ -4,12 +4,16 @@ module Host
       class AttachSubscriptions < Actions::EntryAction
         middleware.use Actions::Middleware::KeepCurrentUser
 
-        def plan(host, pools_with_quantities)
+        def plan(host, pools_with_quantities_params)
           action_subject(host)
-
           sequence do
             pool_ids = []
+            pools_with_quantities = pools_with_quantities_params.map do |pool_with_quantity|
```

Comment:
When I converted this action to Async bulk actions I needed to change the input params to hash. It wont accept raw objects. So I added a to_hash/fetch methods to PoolWithQuantity and used that to get back the original object